spatial-utilities/src/main/java/org/h2gis/utilities/JDBCUtilities.java

```diff
@@ -278,6 +304,58 @@ public static boolean tableExists(Connection connection, String tableName) throw
     }
 
     /**
+     * Returns the list of table names.
+     *
+     * @param metaData Database meta data
+     * @param catalog A catalog name. Must match the catalog name as it is stored in the database.
+     *                      "" retrieves those without a catalog; null means that the catalog name should not be used to
+     *                      narrow the search
+     * @param schemaPattern A schema name pattern. Must match the schema name as it is stored in the database.
+     *                      "" retrieves those without a schema.
+     *                      null means that the schema name should not be used to narrow the search
+     * @param tableNamePattern A table name pattern. Must match the table name as it is stored in the database
+     * @param types A list of table types, which must be from the list of table types returned from getTableTypes(),
+     *              to include. null returns all types
+     * @return The integer primary key used for edition[1-n]; 0 if the source is closed or if the table has no primary
+     *         key or more than one column as primary key
+     */
+    public static List<String> getTableNameList(DatabaseMetaData metaData, String catalog, String schemaPattern,
+                                                 String tableNamePattern, String [] types) throws SQLException {
+        List<String> tableList = new ArrayList<String>();
+        ResultSet rs = metaData.getTables(catalog, schemaPattern, tableNamePattern, types);
+        try {
+            while (rs.next()) {
+                tableList.add(rs.getString("TABLE_NAME"));
+            }
+        } finally {
+            rs.close();
+        }
+        return tableList;
+    }
+
+    /**
+     * Returns the list of distinct values contained by a field from a table from the database
+     *
+     * @param connection Connection
+     * @param tableName Name of the table containing the field.
+     * @param fieldName Name of the field containing the values.
+     * @return The list of distinct values of the field.
+     */
+    public static List<String> getFieldValueList(Connection connection, String tableName, String fieldName) throws SQLException {
+        final Statement statement = connection.createStatement();
+        List<String> fieldValues = new ArrayList<String>();
+        try {
+            ResultSet result = statement.executeQuery("SELECT DISTINCT "+fieldName+" FROM "+TableLocation.parse(tableName));
```

Comment:
For the field name use TableLocation.quoteIdentifier before including it in a query.